upgrades/tests/test_cluster.py

```diff
@@ -0,0 +1,163 @@
+import pytest
+import unittest.mock as mock
+
+import upgrade_testing_framework.cluster_management.cluster as cluster
+
+def test_WHEN_create_Cluster_AND_not_enough_nodes_THEN_raises():
+    # Set up test
+    test_cluster_name = "cluster-name"
+    test_num_nodes = 0 # must be > 1
+    mock_docker_client = mock.Mock()
+    test_image = "image"
+
+    # Run our test
+    with pytest.raises(ValueError):
+        cluster.Cluster(test_cluster_name, test_num_nodes, mock_docker_client, test_image)
+
```

Comment:
nitpick: I suspect flake8 is going to suggest that there should be 2 lines between each of these (https://peps.python.org/pep-0008/#blank-lines) -- maybe try running the auto-transform thing on this? Or just wait for it to complain 🙃